.github/workflows/atfe_nightly_build_and_test.yml

```diff
@@ -0,0 +1,58 @@
+# This workflow runs all build scripts in parallel on Linux.
+# It is intended to be triggered as part of a **nightly build** to
+# validate all build configurations and test them automatically.
+#
+# Each build script is paired with a corresponding test script using
+# a matrix strategy. Each build+test pair runs as a separate job,
+# allowing for concurrency and clear traceability of failures.
+
+name: ATfE Nightly Build and Test
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * *'  # Every day at 02:00 UTC
+
+jobs:
+  build-and-test-toolchain:
+    name: Build ${{ matrix.build_script }} + Test ${{ matrix.test_script }} on ${{ matrix.runner }}
+    runs-on: ${{ matrix.runner }}
+
+    strategy:
+      fail-fast: false  # Prevents one job failure from cancelling all
+      matrix:
+        include:
+          - build_script: build.sh
+            test_script: test.sh
+            runner: ubuntu-22.04
+          - build_script: build_llvmlibc_overlay.sh
+            test_script: test.sh
```

Comment:
We do not test overlay packages yet - will the test script exit cleanly?

Reply:
Indeed, given the execution flow for the overlay packages is slightly different I think it would be better to have a separate workflow to deal with them.
The matrix configuration can then be used to handle the different OSs we need to cover.